pkg/exasol-rest-api/integration_test.go

```diff
@@ -0,0 +1,207 @@
+}
+
+func (suite *IntegrationTestSuite) SetupSuite() {
+	suite.ctx = context.Background()
+	suite.defaultExasolUsername = "api_service_account"
+	suite.defaultExasolPassword = "secret_password"
+	suite.exasolContainer = runExasolContainer(suite.ctx)
+	suite.exasolPort = getExasolPort(suite.exasolContainer, suite.ctx)
+	createDefaultServiceUserWithAccess(suite.defaultExasolUsername, suite.defaultExasolPassword, suite.exasolPort)
+}
+
+func (suite *IntegrationTestSuite) startServer(application exasol_rest_api.Application) *gin.Engine {
+	gin.SetMode(gin.TestMode)
+	router := gin.Default()
+	router.GET("/api/v1/query/:query", application.Query)
+	suite.appProperties = application.Properties
+	return router
+}
+
+func (suite *IntegrationTestSuite) TestQuery() {
+	router := suite.startServer(suite.createApplicationWithDefaultProperties())
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/query/SELECT * FROM TEST_SCHEMA_1.TEST_TABLE", nil)
+	onError(err)
+	responseRecorder := httptest.NewRecorder()
+	router.ServeHTTP(responseRecorder, req)
+	suite.Equal(http.StatusOK, responseRecorder.Code)
+	suite.Equal("{\"status\":\"ok\",\"responseData\":{\"results\":[{\"resultType\":\"resultSet\",\"resultSet\":{\"numColumns\":2,\"numRows\":1,\"numRowsInMessage\":1,\"columns\":[{\"name\":\"X\",\"dataType\":{\"type\":\"DECIMAL\",\"precision\":18,\"scale\":0}},{\"name\":\"Y\",\"dataType\":{\"type\":\"VARCHAR\",\"size\":100,\"characterSet\":\"UTF8\"}}],\"data\":[[15],[\"test\"]]}}],\"numResults\":1}}",
+		responseRecorder.Body.String())
+}
+
+func (suite *IntegrationTestSuite) TestInsertNotAllowed() {
+	router := suite.startServer(suite.createApplicationWithDefaultProperties())
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/query/CREATE SCHEMA not_allowed_schema", nil)
+	onError(err)
+	responseRecorder := httptest.NewRecorder()
+	router.ServeHTTP(responseRecorder, req)
+	suite.Equal(http.StatusOK, responseRecorder.Code)
+	suite.Contains(responseRecorder.Body.String(),
+		"{\"status\":\"error\",\"exception\":{\"text\":\"insufficient privileges for creating schema")
+}
+
+func (suite *IntegrationTestSuite) TestExasolUserWithoutCreateSessionPrivilege() {
+	username := "user_without_session"
+	password := "secret"
+	suite.createExasolUser(username, password)
+	router := suite.startServer(suite.createApplication(&exasol_rest_api.ApplicationProperties{
+		ExasolUser:                username,
+		ExasolPassword:            password,
+		ExasolPort:                suite.exasolPort,
+		ExasolWebsocketApiVersion: 2,
+	}))
+
+	req, err := http.NewRequest(http.MethodGet, "/api/v1/query/SELECT * FROM TEST_SCHEMA_1.TEST_TABLE", nil)
+	onError(err)
+	responseRecorder := httptest.NewRecorder()
+	router.ServeHTTP(responseRecorder, req)
+	suite.Equal(http.StatusBadRequest, responseRecorder.Code)
+	suite.Contains(responseRecorder.Body.String(),
+		"{\"ErrorCode\":\"EXA-REST-API-1\",\"Message\":\"[08004] Connection exception - insufficient privileges: CREATE SESSION.\"}")
```

Comment:
It's not clear to me which user has insufficient privileges here.
```suggestion
		"{\"ErrorCode\":\"EXA-REST-API-1\",\"Message\":\"[08004] Connection exception - The exasol user 'user name' had insufficient privileges: CREATE SESSION.\"}")
```
Add a mitigation.

Reply:
This is also the error from the database, not ours. We can try wrapping them or something, but not in this pull request